src/mesh.cpp

```diff
@@ -220,20 +220,55 @@ void UnstructuredMesh::to_hdf5(hid_t group) const
   write_dataset(mesh_group, "type", mesh_type);
   write_dataset(mesh_group, "filename", filename_);
   write_dataset(mesh_group, "library", this->library());
-  // write volume of each element
-  vector<double> tet_vols;
-  xt::xtensor<double, 2> centroids({static_cast<size_t>(this->n_bins()), 3});
+
+  if (specified_length_multiplier_)
+    write_dataset(mesh_group, "length_multiplier", length_multiplier_);
```

Comment:
Since you're adding things to the HDF5 file, you'll need to update our io_formats documentation: `docs/source/io_formats/statepoint.rst`